Replace Rufus-scheduler with Heroku scheduler

JayKay24 opened the pull request:
#### What does this PR do?

This PR adds replacements for the Rufus cron scheduler with Heroku scheduler.

#### Description
Currently, Rufus-scheduler remains inactive as soon as the Heroku
servers go idle, thus being rendered unable to send the links to
registered users.

This addition replaces the current implementation if Rufus-scheduler with
Heroku scheduler.

JayKay24 committed:
chore(Heroku scheduler): Replace Rufus-scheduler with Heroku scheduler

Currently, rufus-scheduler remains inactive as soon as the Heroku
servers go idle, thus being rendered unable to send the links to
registered users.

This commit replaces the current implementation if Rufus-scheduler with
Heroku scheduler.

[Delivers #159458306]

morriskimani commented on app/controllers/link_controller.rb:
```diff
@@ -10,7 +10,7 @@
   bing_search_api = BingService::BingWebSearchApi.new
 
-  get '/category_links/:category_id' do
+  get '/category/:category_id/links' do
```
I like the new restful routes. 
Just a small issue: I'd have expected `category` to be plural. Check out the link that I shared on RESTful routes.

morriskimani commented on spec/models/category_model_spec.rb:
```diff
@@ -3,29 +3,18 @@
-               category_name: 'Superman',
-               description: 'A world of heroes'
-             )).to be_valid
+      should validate_presence_of :category_name
```
No need to change this one now... But in future, prefer `expect` rather than `should`. I think `should` has actually been deprecated. See [this on reasons why](https://github.com/rspec/rspec-expectations/blob/master/Should.md)

morriskimani commented on jobs/link_sender.rb:
```diff
@@ -3,57 +3,61 @@
-        until link_name.nil?
-          result_set.shuffle!
+module LinkSender
+  def send_links
```
IMO, this method is too long. Which means that it's doing too much work. Remember **Single Responsibility Principle**...
Several things are happening here:
- Configuring Pony
- Finding a link that hasn't been sent yet
- Saving and sending the link.
etc.

morriskimani commented on Rakefile:
```diff
@@ -3,11 +3,20 @@ require 'sinatra/activerecord/rake'
 
-Coveralls::RakeTask.new
\ No newline at end of file
+namespace :mailer do
```
👍

morriskimani reviewed (commented):
A marked improvement on tests. Testing for model validations and more edge cases.

The only major issue on the PR is the `send_links` method which I have flagged. Other than that, looks good!

JayKay24 committed:
chore(refactor): Apply SRP to send_links method

This commit refactors the send_links method to
conform to the Single Responsibility Principle
by moving some logic into their own helper methods.
